src/core.rs

```diff
@@ -431,11 +431,97 @@ impl IndexMut<usize> for UeidType {
     }
 }
 
+// ObjectIdentifier is an identifier mechanism defined by [ITU
+// X.660](https://www.itu.int/rec/T-REC-X.660-201107-I/en) for naming objects in a
+// globally-unambiguous way. It is a sequence of integer nodes (textually represented as
+// dot-delimited integers) representing a path through the OID tree. Each node in the tree is
+// controlled by an assigning authority.
+#[derive(Debug, PartialEq, Eq, Clone, PartialOrd, Ord)]
+pub struct ObjectIdentifier(Bytes);
```

Comment:
This crate is using `derive_more`, so it is very likely you could simplify this to doing:

```suggestion
#[derive(Debug, PartialEq, Eq, Clone, PartialOrd, Ord, TryFrom, AsRef, AsMut, Deref, DerefMut)]
pub struct ObjectIdentifier(Bytes);
```

Reply:
It give me "`TryFrom` connot be derived for structs" but I've added the other derivations.